sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/domain/domain.scala

```diff
@@ -142,12 +146,37 @@ final case class DisclosedContract(
     driverMetadata: Bytes,
     transactionVersion: TransactionVersion,
 ) {
-  def toLf: LfDisclosedContract =
-    LfDisclosedContract(
-      templateId,
-      contractId,
-      argument,
-      keyHash,
+
+  private[this] def keyWithWithMaintainer =
+    (keyValue, keyMaintainers) match {
+      case (None, None) => None
+      case (Some(value), Some(maintainers)) =>
+        Some(GlobalKeyWithMaintainers.assertBuild(
+          templateId,
+          value,
+          maintainers,
+          packageName
+        )
+        )
+      case _ =>
+        throw new Error("unexpected mismatch between keyValue and keyMaintainers")
+    }
+
+  def toLf =
+    LfFatContractInstance.fromCreateNode(
+      LfNode.Create(
```

Comment:
This complication goes away when a `com.digitalasset.canton.ledger.api.domain.DisclosedContract` goes away.